concrete-csprng/src/counter/mod.rs

```diff
@@ -335,48 +365,107 @@ impl<G: AesBatchedGenerator> AesCtrGenerator<G> {
         }
         let output = (0..n_child.0)
             .into_par_iter()
-            .map(self.get_creation_closure(child_bytes));
+            .map(self.get_sequential_creation_closure(child_bytes));
         self.regenerate_batch(n_child, child_bytes);
         Some(output)
     }
 
+    /// Tries to perform an alternate fork of the current generator into `n_child` generators
+    /// each able to yield as many random bytes as wanted.
+    pub fn try_alternate_fork(
+        &mut self,
+        n_child: ChildCount,
+    ) -> Option<impl Iterator<Item = AesCtrGenerator<G>>> {
+        let output = (0..n_child.0).map(self.get_alternate_creation_closure(n_child));
+        self.update_counter_step(n_child);
+        Some(output)
+    }
+
+    /// Tries to perform an alternate fork of the current generator into `n_child` generators each
+    /// able to yield as many random bytes as wanted, returned into a parallel iterator.
+    ///
+    /// # Notes
+    ///
+    /// This method necessitate the "multithread" feature.
+    #[cfg(feature = "multithread")]
+    pub fn par_try_alternate_fork(
+        &mut self,
+        n_child: ChildCount,
+    ) -> Option<impl IndexedParallelIterator<Item = AesCtrGenerator<G>>>
+    where
+        G: Send + Sync,
+    {
+        let output = (0..n_child.0)
+            .into_par_iter()
+            .map(self.get_alternate_creation_closure(n_child));
+        self.update_counter_step(n_child);
+        Some(output)
+    }
+
     fn regenerate_batch(&mut self, n_child: ChildCount, child_bytes: BytesPerChild) {
         let generate = self.state.shift(child_bytes.0 * n_child.0);
         if let ShouldGenerateBatch::GenerateBatch = generate {
             self.batch = self.generator.generate_batch(self.state.get_aes_counter());
         }
     }
 
+    fn update_counter_step(&mut self, n_child: ChildCount) {
+        self.state.shift(n_child.0 * self.step.0 as usize);
```

Comment:
Shouldn't this function generate a new batch of randomness if the output is ShouldGenerateBatch::GenerateBatch ? See regenerate_batch.